query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryParser.scala

```diff
@@ -62,7 +62,7 @@ class QueryParser(analyzer: SoQLAnalyzer[SoQLType], schemaFetcher: SchemaFetcher
     def toUserColumnId(columnName: ColumnName) = columnName.name
 
     analyses match {
-      case PipeQuery(l, r) =>
+      case PipeQuery(l, r, inParen) =>
```

Comment:
Ew, the whole point of the binary tree structure is that parens are implicit in the shape of the tree.  Passing this around is horribly ugly.  No.

Reply:
Both Select a UNION Select b INTERSECT Select c and
         (Select a UNION Select b) INTERSECT Select c generates the same binary tree {a, {b}}, c}.  The alternative is probably generate something similar to function call?

Reply:
Then that's a bug in the parser and should be fixed there.

Reply:
Ok?  That's basically the same as saying "a op b otherop c" generates the same tree as "(a op b) otherop c", which is fine as long as "op" has a higher precedence than "otherop".  If that's not the case then it's a bug in the parser and should be fixed there.